diff --git a/oss/build_whl.sh b/oss/build_whl.sh
index 275c868..f312188 100755
--- a/oss/build_whl.sh
+++ b/oss/build_whl.sh
@@ -27,11 +27,15 @@ function main() {
   [ -e .bazelrc ] && rm .bazelrc
 
   write_to_bazelrc "build -c opt"
+  write_to_bazelrc "build --action_env MACOSX_DEPLOYMENT_TARGET=11.0"
   write_to_bazelrc "build --cxxopt=-std=c++17"
   write_to_bazelrc "build --host_cxxopt=-std=c++17"
-  write_to_bazelrc "build --linkopt=\"-lrt -lm\""
   write_to_bazelrc "build --experimental_repo_remote_exec"
-  write_to_bazelrc "build --python_path=\"${PYTHON_BIN}\""
+  write_to_bazelrc "common --python_path=\"${PYTHON_BIN}\""
+  PLATFORM="$(uname)"
+  if [[ "$PLATFORM" != "Darwin" ]]; then
+    write_to_bazelrc "build --linkopt=\"-lrt -lm\""
+  fi
 
   if [ -n "${CROSSTOOL_TOP}" ]; then
     write_to_bazelrc "build --crosstool_top=${CROSSTOOL_TOP}"
@@ -42,8 +46,8 @@ function main() {
   # https://github.com/bazelbuild/bazel/issues/8622
   export USE_BAZEL_VERSION=5.4.0
   bazel clean
-  bazel build ...
-  bazel test --verbose_failures --test_output=errors ...
+  bazel build ... --action_env PYTHON_BIN_PATH="${PYTHON_BIN}"
+  bazel test --verbose_failures --test_output=errors ... --action_env PYTHON_BIN_PATH="${PYTHON_BIN}"
 
   DEST="/tmp/array_record/all_dist"
   # Create the directory, then do dirname on a non-existent file inside it to
@@ -71,7 +75,11 @@ function main() {
 
   pushd ${TMPDIR}
   echo $(date) : "=== Building wheel"
-  ${PYTHON_BIN} setup.py bdist_wheel --python-tag py3${PYTHON_MINOR_VERSION}
+  plat_name=""
+  if [[ "$(uname)" == "Darwin" ]]; then
+    plat_name="--plat-name macosx_11_0_universal2"
+  fi
+  ${PYTHON_BIN} setup.py bdist_wheel --python-tag py3${PYTHON_MINOR_VERSION} $plat_name
 
   if [ -n "${AUDITWHEEL_PLATFORM}" ]; then
     echo $(date) : "=== Auditing wheel"
